scripts/create_x509_chain_crypto.py

```diff
@@ -0,0 +1,558 @@
+        )
+        device_csr = create_csr(
+            private_key=device_private_key,
+            csr_file=device_csr_file,
+            subject=device_common_name + str(i),
+            is_ca=False,
+        )
+
+        builder = create_cert_builder(
+            subject=device_csr.subject,
+            issuer_name=inter_cert_subject,
+            public_key=device_csr.public_key(),
+            days=int(days / 100),
+            is_ca=False,
+        )
+
+        device_cert = builder.sign(
+            private_key=inter_key, algorithm=hashes.SHA256(), backend=default_backend()
+        )
+        with open(device_cert_file, "wb") as f:
+            f.write(device_cert.public_bytes(serialization.Encoding.PEM))
+
+
+def create_verification_cert(nonce, issuer_password, root_verify=False, key_size=4096):
+    encoded_issuer_password = str.encode(issuer_password)
+
+    if root_verify:
+        verification_password_file = "demoCA/private/verfiication_ca_key.pem"
+        verfication_csr_file = "demoCA/newcerts/verfiication_ca_csr.pem"
+        verfication_cert_file = "demoCA/newcerts/verfiication_ca_cert.pem"
+        issuer_key_file = "demoCA/private/ca_key.pem"
+        issuer_cert_file = "demoCA/newcerts/ca_cert.pem"
+    else:
+        verification_password_file = "demoCA/private/verfiication_inter_key.pem"
+        verfication_csr_file = "demoCA/newcerts/verfiication_inter_csr.pem"
+        verfication_cert_file = "demoCA/newcerts/verfiication_inter_cert.pem"
+        issuer_key_file = "demoCA/private/intermediate_key.pem"
+        issuer_cert_file = "demoCA/newcerts/intermediate_cert.pem"
+
+    with open(issuer_cert_file, "rb") as fh:
+        pem_data = fh.read()
+        issuer_cert = x509.load_pem_x509_certificate(pem_data, default_backend())
+
+    with open(issuer_key_file, "rb") as key_file:
+        pem_data = key_file.read()
+        issuer_private_key = serialization.load_pem_private_key(
+            pem_data, password=encoded_issuer_password, backend=default_backend()
+        )
+
+    verification_private_key = create_private_key(
+        password_file=verification_password_file, password=None, key_size=key_size
+    )
+    verification_csr = create_csr(
+        private_key=verification_private_key,
+        csr_file=verfication_csr_file,
+        subject=nonce,
+        is_ca=False,
+    )
+
+    verification_builder = create_cert_builder(
+        subject=verification_csr.subject,
+        issuer_name=issuer_cert.subject,
+        public_key=verification_csr.public_key(),
+    )
+
+    verification_cert = verification_builder.sign(
+        private_key=issuer_private_key, algorithm=hashes.SHA256(), backend=default_backend()
+    )
+    with open(verfication_cert_file, "wb") as f:
+        f.write(verification_cert.public_bytes(serialization.Encoding.PEM))
+
+
+def create_cert_builder(subject, issuer_name, public_key, days=30, is_ca=False):
```

Comment:
This one is 30.  Hmmm.  I'm only paying attention to the days because the inconsistency stands out, but also because cert expiration is an important part of security.  If this is only for tests, why do we have any certs that last longer than one day?  If this is for other uses, then maybe we need to pay extra attention because certs that last 10 years seem dangerously long-lived.

Reply:
changed to 365...the default of everything is 365 or 1 year....there is no 10 year one now...also intermediates get crated for almost a month and device ones get created for some days

---
In reply to: [432557761](https://github.com/Azure/azure-iot-sdk-python/pull/574#discussion_r432557761) [](ancestors = 432557761)